agent/client.go

```diff
@@ -125,16 +125,23 @@ func (c *accessKeyClient) GetUserCredentials() error {
 	options := &sts.GetSessionTokenInput{
 		DurationSeconds: &durationSeconds,
 	}
+
+	log.Debug("accessKeyClient.GetUserCredentials: calling sts.GetSessionToken()")
+
 	response, err := c.sts.GetSessionToken(options)
+	if err != nil {
+		log.Errorf("accessKeyClient.GetUserCredentials: Failed to get session token: %v", err.Error())
+		return err
+	}
+
+	log.Debug("accessKeyClient.GetUserCredentials: Successfully called sts.GetSessionToken()")
+
 	creds := &sts.Credentials{
 		AccessKeyId:     response.Credentials.AccessKeyId,
 		SessionToken:    response.Credentials.SessionToken,
 		SecretAccessKey: response.Credentials.SecretAccessKey,
 		Expiration:      response.Credentials.Expiration,
 	}
-	if err != nil {
```

Comment:
this is a conflict with base: i moved this err check up ABOVE the creds assignment which used the response from GetSessionToken (regardless of error)... sigh